packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNodeActions.tsx

```diff
@@ -176,7 +178,41 @@ const TreeSingleNodeActions: FC<Props> = ({
                     </Menu.Item>
                 )}
 
-                {customMetrics.length > 0 && isDimension(item) ? (
+                {isCustomDimension(item) && (
+                    <>
+                        <Menu.Item
+                            component="button"
+                            icon={<MantineIcon icon={IconEdit} />}
+                            onClick={(e) => {
+                                e.stopPropagation();
+                                toggleCustomDimensionModal({
+                                    item,
+                                    isEditing: true,
+                                });
+                            }}
+                        >
+                            Edit custom dimension
+                        </Menu.Item>
+                        <Menu.Item
+                            color="red"
+                            component="button"
+                            icon={<MantineIcon icon={IconTrash} />}
+                            onClick={(e) => {
+                                e.stopPropagation();
+                                removeCustomDimension(
+                                    getCustomDimensionId(item),
+                                );
+                            }}
+                        >
+                            Remove custom dimension
+                        </Menu.Item>
+                    </>
+                )}
+
+                {customMetrics.length > 0 &&
+                (isDimension(item) ||
+                    (isCustomDimension(item) &&
```

Comment:
show custom metric submenu for customDimensions

Reply:
Not sure why we need all these checks when `customMetrics` is going to be empty in those cases. Should we just have `customMetrics.length > 0` ?

Reply:
ok, we need the `item` typing to access properties. Changed this to use `isCustomSqlDimension`.